libsolutil/CommonData.h

```diff
@@ -93,10 +93,8 @@ template <class T>
 inline std::vector<T> operator+(std::vector<T>&& _a, std::vector<T>&& _b)
 {
 	std::vector<T> ret(std::move(_a));
-	if (&_a == &_b)
-		ret += ret;
-	else
-		ret += std::move(_b);
+	assert(&_a != &_b);
```

Comment:
In the other version it was not possible to call this if `T` was not copyable.

Reply:
Is there no solution that works in all cases?